index.js

```diff
@@ -115,14 +28,14 @@ PayshieldCodec.prototype.init = function(config) {
     var commandsObj = merge({}, defaultFormat, config.messageFormat);
 
     if (this.headerPattern === false) {
-        throw new Error('Cant parse header pattern!');
+        throw this.errors.parser({cause: 'Cant parse header pattern!'});

```

Comment:
its not working, that's why im passing cause

Reply:
Still this is not a correct way and it is not working properly too. `cause` is meant to be an Error object. I think there is a bug in ut-error: when you define an error with default message, it cannot be overridden by passing it as string (or any other way), when calling the error function.

In addition, the better way to implement this is to define the error with a default message that may include placeholders for parameters. This will make all strings available to the error API and make it easier for potential localization. This will also workaround this bug.

@stambata, can you have a look? We may even entirely disable passing strings when creating the errors and deprecate (and later break) all such bad patterns.

Reply:
@kalinkrustev 
> I think there is a bug in ut-error: when you define an error with default message, it cannot be overridden by passing it as string (or any other way), when calling the error function.

That's correct. Currently if there's a defaultMessage it is set as a message no matter what. Should I change it in a way to use the defaultMessage only when there's no message?

> We may even entirely disable passing strings when creating the errors and deprecate (and later break) all such bad patterns.

Sure we can.

Let's discuss here: https://github.com/softwaregroup-bg/ut-error/issues/5